File: c++/83_Remove_Duplicates_from_Sorted_List.cpp
#include <iostream>
#include <vector>
#include <map>
#include <string>

using namespace std;

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */


class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head)
            return head;
        ListNode *p = head;
        while (p->next){
            if (p->next->val == p->val)
                p->next = p->next->next;
            else
                p = p->next;
        }
        return head;
        
        
    }
};


int main() {



}

File: python/83_Remove_Duplicates_from_Sorted_List.py
#!/usr/bin/python
import sys
import os

class Solution(object):
    def deleteDuplicates(self, head):
        if head == None:
            return head
        tmp = head
        while tmp.next:
            if tmp.next.val == tmp.val:
                tmp.next = tmp.next.next
            else:
                tmp = tmp.next
        return head





def main():
    s = Solution()

if __name__ == "__main__":
    main()
